pyo3-macros-backend/src/method.rs

```diff
@@ -351,52 +332,70 @@ impl<'a> FnSpec<'a> {
                 *python_name = Some(syn::Ident::new("__new__", Span::call_site()));
-                if matches!(fn_type_attr, Some(MethodTypeAttribute::New)) {
+                if matches!(method_attributes.as_slice(), [MethodTypeAttribute::New(_)]) {
                     (FnType::FnNew, false, Some(CallingConvention::TpNew))
                 } else {
                     (FnType::FnNewClass, true, Some(CallingConvention::TpNew))
                 }
             }
-            Some(MethodTypeAttribute::ClassMethod) => (FnType::FnClass, true, None),
-            Some(MethodTypeAttribute::Getter) => {
-                // Strip off "get_" prefix if needed
-                if python_name.is_none() {
+            [MethodTypeAttribute::ClassMethod(_)] => (FnType::FnClass, true, None),
+            [MethodTypeAttribute::Getter(_, name)] => {
+                if let Some(name) = name.take() {
+                    ensure_spanned!(
+                        python_name.replace(name).is_none(),
+                        python_name.span() => "`name` may only be specified once"
+                    );
+                } else if python_name.is_none() {
+                    // Strip off "get_" prefix if needed
                     *python_name = strip_fn_name("get_");
                 }
 
                 (
-                    FnType::Getter(parse_receiver("expected receiver for #[getter]")?),
+                    FnType::Getter(parse_receiver("expected receiver for `#[getter]`")?),
                     true,
                     None,
                 )
             }
-            Some(MethodTypeAttribute::Setter) => {
-                // Strip off "set_" prefix if needed
-                if python_name.is_none() {
+            [MethodTypeAttribute::Setter(_, name)] => {
+                if let Some(name) = name.take() {
+                    ensure_spanned!(
+                        python_name.replace(name).is_none(),
+                        python_name.span() => "`name` may only be specified once"
+                    );
+                } else if python_name.is_none() {
+                    // Strip off "get_" prefix if needed
                     *python_name = strip_fn_name("set_");
                 }
 
                 (
-                    FnType::Setter(parse_receiver("expected receiver for #[setter]")?),
+                    FnType::Setter(parse_receiver("expected receiver for `#[setter]`")?),
                     true,
                     None,
                 )
             }
-            None => (
-                FnType::Fn(parse_receiver(
-                    "static method needs #[staticmethod] attribute",
-                )?),
-                true,
-                None,
-            ),
+            [first, second, ..] => {
+                bail_spanned!(
+                    second.span() =>
+                    format!("`#[{}]` may not be combined with `#[{}]`", first.ident(), second.ident())
+                )
+            }
```

Comment:
nit: this should list all the attributes to avoid a confusing error for
```rust
#[new]
#[classmethod] 
#[whatever]
fn method() {}
```

Reply:
Agreed, will adjust.